apis/delivery/deliveries.py

```diff
@@ -1,15 +1,83 @@
-from flask import request, g, current_app
+from flask import request, g, current_app, send_file
 from flask_restplus import Namespace, Resource
-from flask_jwt_extended import jwt_required, decode_token
+from flask_jwt_extended import jwt_required, decode_token, get_jwt_identity
 from core import constants as C
+from core.protocol import get_delivery, listing_accessed, delivered
+from .parsers import delivery_parser
 
 api = Namespace('Delivery', description='Delivery endpoint for requesting purchased payloads')
 
-@api.route('/', methods=['GET','POST'])
+@api.route('/', methods=['GET'])
 class Delivery(Resource):
+    @api.expect(delivery_parser)
+    @api.response(200, C.CONTENT_DELIVERED)
+    @api.response(401, C.LOGIN_FAILED)
+    @api.response(412, C.INSUFFICIENT_PURCHASED)
     @jwt_required
     def get(self):
         """
-        Stub to verify required JWT tokens
+        Return the listing requested as a delivery object
         """
-        return dict(message="Huzzah!"), 200
\ No newline at end of file
+        args = delivery_parser.parse_args()
+        delivery_hash = args['delivery_hash']
+        owner = get_jwt_identity()
+        current_app.logger.info(f'Retrieving {delivery_hash} for delivery')
+        authorized_delivery = get_delivery(delivery_hash)
```

Comment:
this is were i was talking about. I _think_ this can just be:

    delivery_owner, requested_bytes, delivered_bytes = get_delivery(delivery_hash)